### PR TITLE
RD-5371 auto instrument load handlers just as aws

### PR DESCRIPTION
**Pull request by saartochner**
Instead of parse the handler's string ourselves, I copied the parsing code from AWS official js lambda API: 
https://github.com/aws/aws-lambda-nodejs-runtime-interface-client

### CODE REVIEW

File: auto-instrument-handler/index.test.js [hunk 1]
```diff
@@ -3,47 +3,38 @@ const index = require('./index');
 describe('tracer', () => {
   test('handler call the original function', () => {
     process.env[index.LUMIGO_SWITCH_OFF] = 'TRUE';
+    process.env.LAMBDA_TASK_ROOT = '/var/task';
 
     process.env[index.ORIGINAL_HANDLER_KEY] = '';
     expect(() => index.handler({}, {})).toThrowError(
-      'Could not load the original handler. Are you sure that the handler is correct?'
+      'Bad handler'
```
Comment (orr-levinger) [inline auto-instrument-handler/index.test.js:10]: why change the original message?

File: auto-instrument-handler/aws/aws-user-function.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+}
+
+/**
+ * Split the handler string into two pieces: the module name and the path to
+ * the handler function.
+ */
+function _splitHandlerString(handler) {
+  let match = handler.match(FUNCTION_EXPR);
+  if (!match || match.length != 3) {
+    throw new MalformedHandlerName("Bad handler");
```
Comment (orr-levinger) [inline auto-instrument-handler/aws/aws-user-function.js:46]: That not an informative message..

File: auto-instrument-handler/index.test.js (hunk 1 above)
Comment (saartochner) [reply auto-instrument-handler/index.test.js:10]: On the contrary - we used to change the original message.
Not we're raising the exact same message as the vanilla runtime :)

File: auto-instrument-handler/aws/aws-user-function.js (hunk 2 above)
Comment (saartochner) [reply auto-instrument-handler/aws/aws-user-function.js:46]: That's not my code 😂 
This is the message that the node runtime throws in this case. I don't want that the message will be different with our instrumentation 🤔

**Review by OrrLumigo — APPROVED** [review]

File: auto-instrument-handler/aws/Errors.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
+/**
```
Comment (saartochner) [inline auto-instrument-handler/aws/Errors.js:1]: This file is a copied file

File: auto-instrument-handler/aws/aws-user-function.js [hunk 4]
```diff
@@ -0,0 +1,157 @@
+/**
```
Comment (saartochner) [inline auto-instrument-handler/aws/aws-user-function.js:1]: This file is a copied file

**Review by uri-p — APPROVED** [review]

**Pull request merged**